core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -104,3 +105,20 @@ def start_board(self, boards: List[Tuple[FlightControllerType, str]]) -> bool:
 
     def restart(self) -> None:
         self.mavlink_manager.restart()
+
+    def _load_endpoints(self) -> None:
+        endpoints = self.settings.get_saved_instances(self.settings.endpoints_store)
```

Comment:
what is the idea behind  sending self.settings.endpoints_store to self.settings ? 
If self.settings has the endpoints_store, why is it necessary to use it as argument ?

Reply:
Because the get_saved_instances function is unaware of what you are trying to get in the first place.

Reply:
why not a simple `self.settings.endpoints` ?
why ArdupiLotManager should keep track of the settings path in the system ?

Reply:
That's actually better.